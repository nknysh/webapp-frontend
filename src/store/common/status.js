import { equals, anyPass } from 'ramda';

export const Status = Object.freeze({
  LOADING: 'LOADING',
  IDLE: 'IDLE',
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS',
  SENDING: 'SENDING',
});

export const isIdle = equals(Status.IDLE);
export const isLoading = equals(Status.LOADING);
export const isError = equals(Status.ERROR);
export const isSuccess = equals(Status.SUCCESS);
export const isSending = equals(Status.SENDING);

export const isActive = anyPass([isLoading, isSending]);
