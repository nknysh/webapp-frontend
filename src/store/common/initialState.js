import { Status } from './status';

export default {
  status: Status.IDLE,
  error: undefined,
  data: undefined,
};
