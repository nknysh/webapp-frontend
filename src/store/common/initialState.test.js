import { Status } from './status';
import initialState from './initialState';

describe('initialState', () => {
  it('returns common initial state', () => {
    expect(initialState).toMatchSnapshot();
  });
});
