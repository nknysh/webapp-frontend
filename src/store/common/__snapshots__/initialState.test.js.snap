// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialState returns common initial state 1`] = `
Object {
  "data": undefined,
  "error": undefined,
  "status": "IDLE",
}
`;
